src/Keter/Context.hs

```diff
@@ -0,0 +1,25 @@
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE DerivingStrategies         #-}
+
+module Keter.Context where
+
+import           Keter.Common
+import           Control.Monad.Trans       (lift)
+import           Control.Monad.IO.Class   (MonadIO)
+import           Control.Monad.IO.Unlift   (MonadUnliftIO)
+import           Control.Monad.Logger      (MonadLogger, MonadLoggerIO, LoggingT(..), runLoggingT)
+import           Control.Monad.Reader      (MonadReader, ReaderT, runReaderT, ask
+                                           , withReaderT)
+
+-- | The top-level keter context monad, carrying around the main logger and some locally relevant configuration structure.
```

Comment:
Please also refer to a relevant blogpost if you have one in mind for this design.
for example https://www.fpcomplete.com/blog/2017/06/readert-design-pattern/ 

this allows newer people to figure out what goes on with all the boilerplate :)

Reply:
I don't think I know of one off the top of my head more specialized than that, so if you don't mind I'll steal that link lol. Added